packages/client/src/pages/ForumPage/index.tsx

```diff
@@ -0,0 +1,304 @@
+import React, { useState } from "react";
+import {
+  Container,
+  Typography,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Paper,
+  TextField,
+  Box,
+  Modal,
+  IconButton,
+} from "@mui/material";
+import { UiButton } from "@/components/ui/UiButton";
+import { UiLayout } from "@/components/ui/UiLayout";
+import FormatBoldIcon from "@mui/icons-material/FormatBold";
+import FormatItalicIcon from "@mui/icons-material/FormatItalic";
+import FormatUnderlinedIcon from "@mui/icons-material/FormatUnderlined";
+import { UiPaper } from "@/components/ui/UiPaper";
+import { paths } from "@/app/constants/paths";
+import { useNavigate } from "react-router-dom";
+import { useAppSelector } from "@/hooks";
+import { RootState } from "@/store";
+import { IProfile } from "@/types/profile.interface";
+
+interface Topic {
+  id: number;
+  title: string;
+  description: string;
+  author: string;
+  messages: number;
+  date: string;
+}
+
+interface CreateTopicModalProps {
+  open: boolean;
+  onClose: () => void;
+  onCreate: (newTopic: Omit<Topic, "id" | "messages" | "date">) => void;
+}
+
+const CreateTopicModal: React.FC<CreateTopicModalProps> = ({ open, onClose, onCreate }) => {
+  const selectProfileInfo = (state: RootState) => state.profile.user as IProfile;
```

Comment:
Приведение типов (`as`) -- это всегда плохо, это как будто мы просто пытаемся обмануть компилятор. В будущем это может привести к проблемам, потому лучше типы описывать явно:

const selectProfileInfo = (state: RootState): IProfile => state.profile.user;
```suggestion
  const selectProfileInfo = (state: RootState): IProfile => state.profile.user;
```

Если же приведение типов используется, чтобы "скрыть" undefined, то это особенно плохо, так как именно этот момент в будущем может сломать весь код. 

1. Поправь, пожалуйста типизацию.
2. Селекторы лучше описывать в отдельном файле `selectors.ts`, потому что так их можно переиспользовать (а я вижу ниже написан точно такой же селектор), ну и плюс это разделение обязанностей, чтобы не было в одном файле всего, что можно.